### PR TITLE
Load user.css in error page protostar

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #25286  .

### Summary of Changes
load user.css

### Testing Instructions
create a user.css in the css folder of protostar
generate a 404

### Expected result
page loads user.css

### Actual result
no reference to user.css

### Documentation Changes Required

none

### CODE REVIEW

**Comment by ghost** [comment]
guess its for #25286

**Comment by brianteeman** [comment]
@franz-wohlkoenig yes - I updated op. created the pr from my phone so easy to mistype

**Comment by Giuse69** [comment]
Maybe adding a check if user.css exists before including it?

**Comment by brianteeman** [comment]
Updated branch to make drone run

**Comment by uGE70** [comment]
I have tested this item :white_check_mark: successfully on 774be4fa6424be8ec7b95d3d7ec14b0f451d50f1<br /><br />user.css is included in 404 error page after applying patch. Thanks.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25289">issues.joomla.org/tracker/joomla-cms/25289</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :red_circle: unsuccessfully on 774be4fa6424be8ec7b95d3d7ec14b0f451d50f1<br /><br />With this PR applied, I get a 404 for the user.css when watching network traffic in developer tools of browser. It needs a check if the file exists before including it. This is different to the index.php and the offline.php, where `JHtml::_('stylesheet', 'user.css' ...` is used, which does that job for us.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25289">issues.joomla.org/tracker/joomla-cms/25289</a>.</sub>

**Comment by uGE70** [comment]
I am a new to these testings, so I might have missed the mark, be patient, but I confirm the user.css is loaded after applying the patch and the site-title class rule I defined as an example is changing the font-size on the 404 error page. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25289">issues.joomla.org/tracker/joomla-cms/25289</a>.</sub>

**Comment by richard67** [comment]
And what if the user.css does not exist, which is the case on a standard Joomla 3.9.x? In this case you get an additional 404.

**Comment by uGE70** [comment]
Understood. I also get a second 404 when user.css is not included.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25289">issues.joomla.org/tracker/joomla-cms/25289</a>.</sub>

**Comment by brianteeman** [comment]
made a small change - please retest

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on c51af2af6130382bb6b1953404918ec0547164a7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25289">issues.joomla.org/tracker/joomla-cms/25289</a>.</sub>

**Comment by uGE70** [comment]
I have tested this item :white_check_mark: successfully on c51af2af6130382bb6b1953404918ec0547164a7<br /><br />If user.css is included, the css rules are successfully added to the 404 page.
If user.css is missing, it doesn't generate another 404 error code.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25289">issues.joomla.org/tracker/joomla-cms/25289</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on c51af2af6130382bb6b1953404918ec0547164a7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25289">issues.joomla.org/tracker/joomla-cms/25289</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25289">issues.joomla.org/tracker/joomla-cms/25289</a>.</sub>

File: templates/protostar/error.php [hunk 1]
```diff
@@ -68,6 +68,9 @@
 		</style>
 	<?php endif; ?>
 	<link href="<?php echo $this->baseurl; ?>/templates/<?php echo $this->template; ?>/css/template.css" rel="stylesheet" />
+	<?php if (@filesize('templates/' . $this->template . '/css/user.css') > 5): ?>
```
Comment (Quy) [inline templates/protostar/error.php:71]:
```suggestion
	<?php if (file_exists('templates/' . $this->template . '/css/user.css')) : ?>
```
Why not use file_exists?

Comment (brianteeman) [reply templates/protostar/error.php:71]: can do if you want. I do it this way for my clients as they often add a user.css file and then remove all the content but dont delete the file.

Comment (Quy) [reply templates/protostar/error.php:71]: I see...so only load it if there is content?

Comment (brianteeman) [reply templates/protostar/error.php:71]: Exactly

Comment (HLeithner) [reply templates/protostar/error.php:71]: Please use file_exists, suppressing the error is really bad and shouldn't be used in core if possible.

Comment (brianteeman) [reply templates/protostar/error.php:71]: It is not suppressing an error at all

Comment (HLeithner) [reply templates/protostar/error.php:71]: Thats what @ does

Comment (Quy) [reply templates/protostar/error.php:71]: The other pages except for the error page link to this file even when empty. Thus it is not consistent on every page.

**Comment by richard67** [comment]
@Quy The other pages use JHTML to add head links while error.php adds them directly. JHTML seems to include a check if the file exists, therefore there no extra check.

**Comment by richard67** [comment]
I think it is by design that the error page doesn’t use JHTML.

**Comment by HLeithner** [comment]
For consistency please use file_exists

**Pull request closed without merging**